mne/morph.py

```diff
@@ -0,0 +1,1182 @@
+        warn_('%s/%s vertices not included in smoothing, consider increasing '
+              'the number of steps'
+              % (len(data_sum) - len(idx_use), len(data_sum)))
+
+    logger.info('    %d smooth iterations done.' % (k + 1))
+
+    data_morphed = maps[nearest, :] * data
+    return data_morphed
+
+
+def _morph_mult(data, e, use_sparse, idx_use_data, idx_use_out=None):
+    """Help morphing.
+
+    Equivalent to "data = (e[:, idx_use_data] * data)[idx_use_out]"
+    but faster.
+    """
+    if len(idx_use_data) < e.shape[1]:
+        if use_sparse:
+            data = e[:, idx_use_data] * data
+        else:
+            # constructing a new sparse matrix is faster than sub-indexing
+            # e[:, idx_use_data]!
+            col, row = np.meshgrid(np.arange(data.shape[1]), idx_use_data)
+            d_sparse = sparse.csr_matrix((data.ravel(),
+                                          (row.ravel(), col.ravel())),
+                                         shape=(e.shape[1], data.shape[1]))
+            data = e * d_sparse
+            data = np.asarray(data.todense())
+    else:
+        data = e * data
+
+    # trim data
+    if idx_use_out is not None:
+        data = data[idx_use_out]
+    return data
+
+
+def _sparse_argmax_nnz_row(csr_mat):
+    """Return index of the maximum non-zero index in each row."""
+    n_rows = csr_mat.shape[0]
+    idx = np.empty(n_rows, dtype=np.int)
+    for k in range(n_rows):
+        row = csr_mat[k].tocoo()
+        idx[k] = row.col[np.argmax(row.data)]
+    return idx
+
+
+def _get_subject_sphere_tris(subject, subjects_dir):
+    spheres = [op.join(subjects_dir, subject, 'surf',
+                       xh + '.sphere.reg') for xh in ['lh', 'rh']]
+    tris = [read_surface(s)[1] for s in spheres]
+    return tris
+
+
+###############################################################################
+# Apply morph to source estimate
+def _get_zooms_orig(morph):
+    """Compute src zooms from morph zooms, morph shape and src shape."""
+    # zooms_to = zooms_from / shape_to * shape_from for each spatial dimension
+    return [mz / ss * ms for mz, ms, ss in
+            zip(morph.zooms, morph.shape, morph.src_data['src_shape_full'])]
+
+
+def _apply_morph_data(morph, stc_from):
+    """Morph a source estimate from one subject to another."""
+    if stc_from.subject is not None and stc_from.subject != morph.subject_from:
+        raise ValueError('stc.subject (%s) != morph.subject_from (%s)'
+                         % (stc_from.subject, morph.subject_from))
+    if morph.kind == 'volume':
+        from dipy.align.reslice import reslice
+
+        # prepare data to be morphed
+        img_to = _interpolate_data(stc_from, morph, mri_resolution=True,
+                                   mri_space=True)
+
+        # reslice to match morph
+        img_to, img_to_affine = reslice(
+            img_to.get_data(), morph.affine, _get_zooms_orig(morph),
+            morph.zooms)
```

Comment:
Since there was the issue with memory brought up: maybe it would make sense to loop over time when doing ``_interpolate_data``, because this whole thing here is anyway only to later re-slice to morph-resolution. Thus it's not 100% necessary to first create a full blown data set with all time points being up-scaled at the same time, right?